cmake/features.cmake

```diff
@@ -77,6 +77,8 @@ ie_dependent_option (ENABLE_TBBBIND_2_5 "Enable TBBBind_2_5 static usage in Open
 ie_dependent_option (ENABLE_INTEL_GNA "GNA support for inference engine" ON
     "NOT APPLE;NOT ANDROID;X86_64;CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 5.4" OFF)
 
+ie_option (ENABLE_INTEL_GNA_DEBUG "GNA debug build" OFF)
```

Comment:
lets add additional check that build type is debug (e.g. CMAKE_BUILD_TYPE STREQUAL "Debug")

Reply:
@mryzhov I would not do it, there is a different define (i guess DEBUG)
if want to limit some code can use something like:
#if defined(GNA_DEBUG) && defied(DEBUG)
#endif

It gives more flexibility to developers

Reply:
Maybe we could call it "GNA diagnostic build" instead of debug to not confuse ourselfs in future

Reply:
@kbruniec didn't get you, I suggested enable extended debug only in the debug build, why do we need it in the release?

Reply:
@mryzhov I can easily imagine some diagnostic features that are useful in release build.
eg performance measurements (which by the way I am doing right now)
especial the ones that need to be done in production environment (eg APO)
so we could provide a binary GNA plugin where some extended diagnostics are enabled.
It would be much easier to exchange official distributed dll with a diagnostic-enabled-release one

Reply:
We can easily manually enable such a diagnostic for the release build if needed, but we should not allow to enable it by common cmake option, because this option enable all of possible diagnostic logs (model dumps, memory dumps, etc)

Reply:
@mryzhov IMO "common cmake option" is perfect for that.
If you want Release build without this diagnostics features you can just not provide this option.
Beside, additional cmake logic to check CMAKE_BUILD_TYPE is not enough for MSVS which is multi-config generator.
I would avoid complicating cmake more than needed.

Having independed (ortogonal) GNA_DEBUG and DEBUG preprocessor definition to use in cpp files givres us more fine-grained controll on the src level.

parts which want to exclude from non DEBUG at all can be guarded by additional
#if defined(DEBUG) && defined(GNA_DEBUG)